src/crimson/osd/osd_operation.h

```diff
@@ -269,14 +282,34 @@ class OperationRegistry {
     static_cast<int>(OperationTypeCode::last_op)
   > op_id_counters = {};
 
+  seastar::timer<seastar::lowres_clock> shutdown_timer;
+  seastar::promise<> shutdown;
 public:
+  OperationRegistry()
+    : shutdown_timer([this] {
+	bool finished = true;
+	for (auto& oplist : registries) {
+	  if (oplist.size()) {
+	    finished = false;
```

Comment:
Maybe (as this type of loop appears multiple times):

const bool finished = std::all_of(registries.begin(), registries.end(), [](const auto& opl) { return !opl.size(); })

Reply:
(will look even better with ranges...)